### ISSUE TITLE
Murmur will not build with newest version of openssl

### ISSUE THREAD

**Issue by EmperorArthur** [mumble-voip/mumble#2626]
See here for what's happening:  https://github.com/openssl/openssl/issues/1604

```
MurmurIce.cpp: In function ‘void impl_Server_updateCertificate(Murmur::AMD_Server_updateCertificatePtr, int, const string&, const string&, const string&)’:
MurmurIce.cpp:1580:19: error: invalid use of incomplete type ‘RSA {aka struct rsa_st}’
  if (BN_cmp(pubRSA->n, privRSA->n) != 0) {
                   ^~
In file included from /usr/include/openssl/rand.h:14:0,
                 from murmur_pch.h:100:
/usr/include/openssl/ossl_typ.h:110:16: note: forward declaration of ‘RSA {aka struct rsa_st}’
 typedef struct rsa_st RSA;
                ^~~~~~
MurmurIce.cpp:1580:31: error: invalid use of incomplete type ‘RSA {aka struct rsa_st}’
  if (BN_cmp(pubRSA->n, privRSA->n) != 0) {
                               ^~
In file included from /usr/include/openssl/rand.h:14:0,
                 from murmur_pch.h:100:
/usr/include/openssl/ossl_typ.h:110:16: note: forward declaration of ‘RSA {aka struct rsa_st}’
 typedef struct rsa_st RSA;
                ^~~~~~
MurmurIce.cpp:1586:19: error: invalid use of incomplete type ‘RSA {aka struct rsa_st}’
  if (BN_cmp(pubRSA->e, privRSA->e) != 0) {
                   ^~
In file included from /usr/include/openssl/rand.h:14:0,
                 from murmur_pch.h:100:
/usr/include/openssl/ossl_typ.h:110:16: note: forward declaration of ‘RSA {aka struct rsa_st}’
 typedef struct rsa_st RSA;
                ^~~~~~
MurmurIce.cpp:1586:31: error: invalid use of incomplete type ‘RSA {aka struct rsa_st}’
  if (BN_cmp(pubRSA->e, privRSA->e) != 0) {
                               ^~
In file included from /usr/include/openssl/rand.h:14:0,
                 from murmur_pch.h:100:
/usr/include/openssl/ossl_typ.h:110:16: note: forward declaration of ‘RSA {aka struct rsa_st}’
 typedef struct rsa_st RSA;
```

**Comment by mkrautz** [comment]
We should definitely try to be compatible with the opaque API of 1.1 if possible.

Not sure if QSsl even supports 1.1 fully yet, though, in any released version.

Will check up on it.

**Comment by EmperorArthur** [comment]
I almost forgot, this issue is directly related to #2621.

**Comment by davidebeatrici** [comment]
Fixed with #2880.
